packages/cosmic-swingset/x/swingset/ibc.go

```diff
@@ -112,99 +112,305 @@ func (ch channelHandler) Receive(ctx *ControllerContext, str string) (ret string
 	}
 }
 
+type channelOpenInitEvent struct {
+	Type              string                    `json:"type"`  // IBC
+	Event             string                    `json:"event"` // channelOpenInit
+	Order             channelexported.Order     `json:"order"`
+	ConnectionHops    []string                  `json:"connectionHops"`
+	PortID            string                    `json:"port"`
+	ChannelID         string                    `json:"channel"`
+	ChannelCapability *capability.Capability    `json:"channelCapability"`
```

Comment:
Note that this capability cannot be marshalled.  It's not necessary, as we "claim" it in the implementation so that we can use the channelID and portID instead.